src/main/java/edu/byu/cs/canvas/CanvasIntegration.java

```diff
@@ -151,19 +153,78 @@ public static void submitGrade(int userId, int assignmentNum, float grade, Strin
     }
 
     /**
-     * Gets the git repository url for the given user from their GitHub Repository assignment submission on canvas
+     * Submits the given grade for the given assignment for the given user
      *
-     * @param userId The canvas user id of the user to get the git repository url for
-     * @return The git repository url for the given user
+     * @requires The maps passed in must support the putIfAbsent method (Map.of() does not)
+     *
+     * @param userId            The canvas user id of the user to submit the grade for
+     * @param assignmentNum     The assignment number to submit the grade for
+     * @param grades            A Map of rubric item id's to grades for that rubric item
+     * @param rubricComments    A Map of rubric item id's to comments to put on that rubric item
+     * @param assignmentComment A comment for the entire assignment, if necessary
      * @throws CanvasException If there is an error with Canvas
      */
-    public static String getGitRepo(int userId) throws CanvasException {
-        CanvasSubmission submission = makeCanvasRequest(
+    public static void submitGrade(int userId, int assignmentNum, Map<String, Float> grades, Map<String, String> rubricComments, String assignmentComment) throws CanvasException {
+        CanvasSubmission submission = getSubmission(userId, assignmentNum);
+        if(submission.rubric_assessment() != null) {
+            for(Map.Entry<String, CanvasIntegration.RubricItem> entry : submission.rubric_assessment().items().entrySet()) {
+                grades.putIfAbsent(entry.getKey(), entry.getValue().points());
+                rubricComments.putIfAbsent(entry.getKey(), entry.getValue().comments());
+            }
+        }
+
+        StringBuilder queryStringBuilder = new StringBuilder();
```

Comment:
It seems like this block of code from line `176` to `191` would do well to be extracted into a separate function. Based on my interpretation of the code as it stands, it seems that a reasonable name could be `buildSubmissionQueryURL()`. It would probably be best to have the function handle **all** of the string building, whereas right now, some is handled in these control blocks, and some occurs via direct string concatenation on line `195`. That would do a lot to improve the readability of the code.